Sources/AppStoreConnectCLI/Commands/TestFlight/BetaTesters/CreateBetaTesterCommand.swift

```diff
@@ -7,80 +7,85 @@ import Foundation
-                return "Invalid input, one or more build Id or beta group name is required when creating a tester"
-
-            case .couldntFindBetaGroup:
-                return "Couldn't find any beta group with input names."
+            case .noGroupsExist(let groupNames, let bundleId):
+                return "One or more of beta groups \"\(groupNames)\" don't exist or don't belong to application with bundle ID \"\(bundleId)\"."
             }
         }
     }
 
     func run() throws {
         let api = try makeClient()
 
-        let request: AnyPublisher<BetaTesterResponse, Error>
-
-        let createWithBuildIds = { ids -> APIEndpoint<BetaTesterResponse> in
-            .create(betaTesterWithEmail: self.email, firstName: self.firstName, lastName: self.lastName, buildIds: ids)
-        }
-
-        let createWithGroupIds = { ids -> APIEndpoint<BetaTesterResponse> in
-            .create(betaTesterWithEmail: self.email, firstName: self.firstName, lastName: self.lastName, betaGroupIds: ids)
-        }
-
-        switch (buildIds, groupNames) {
-            case (let buildIds, _) where !buildIds.isEmpty:
-                let endpoint = createWithBuildIds(buildIds)
-                request = api.request(endpoint).eraseToAnyPublisher()
-
-            case (_, let groupNames) where !groupNames.isEmpty:
-                let endpoint = APIEndpoint.betaGroups(filter: [ListBetaGroups.Filter.name(groupNames)])
-
-                let groupIds = api.request(endpoint).map { $0.data.map(\.id) }
-
-                request = groupIds
-                    .flatMap { (groupIds: [String]) -> AnyPublisher<BetaTesterResponse, Error> in
-                        guard !groupIds.isEmpty else {
-                            return Fail(error: CommandError.couldntFindBetaGroup).eraseToAnyPublisher()
-                        }
+        _ = api
+            // Find app resource id matching bundle Id
+            .appResourceId(matching: bundleId)
+            // Find beta groups matching app resource Id
+            .flatMap {
+                api.request(APIEndpoint.betaGroups(forAppWithId: $0)).eraseToAnyPublisher()
+            }
+            // Check if input group names are belong to the app, else throws error
+            .tryMap { [groups, bundleId] (response: BetaGroupsResponse) -> AnyPublisher<[String], Error> in
+                let groupNamesInApp = Set(response.data.compactMap { $0.attributes?.name })
+                let inputGroupNames = Set(groups)
 
-                        let endpoint = createWithGroupIds(groupIds)
+                guard inputGroupNames.isSubset(of: groupNamesInApp) else {
+                    throw CommandError.noGroupsExist(groupNames: groups, bundleId: bundleId)
+                }
 
-                        return api.request(endpoint).eraseToAnyPublisher()
-                    }
+                return try api.betaGroupIdentifiers(matching: groups)
+            }
+            .flatMap { $0 }
```

Comment:
We should be able to use a flatMap above here and return a `Fail` publisher instead of throwing